pkg/api/validation/validation.go

```diff
@@ -3288,6 +3290,80 @@ func ValidateNamespaceFinalizeUpdate(newNamespace, oldNamespace *api.Namespace)
 	return allErrs
 }
 
+// Set all NodeNames in the subset EndpointAddress slices to nil
+func clearNodeNamesinSubset(subset *api.EndpointSubset) {
+	for n := range subset.Addresses {
+		subset.Addresses[n].NodeName = nil
+	}
+	for n := range subset.NotReadyAddresses {
+		subset.Addresses[n].NodeName = nil
+	}
+}
+
+// ClearReadOnlyEndpointFieldsOnCreate Clears fields that are non user configurable upon create
+func ClearReadOnlyEndpointFieldsOnCreate(endpoints *api.Endpoints) {
+	for i := range endpoints.Subsets {
+		clearNodeNamesinSubset(&endpoints.Subsets[i])
+	}
+}
+
+func buildEndpointAddressToNodeNameMap(ipToNodeName map[string]string, addresses []api.EndpointAddress) {
+	// Construct lookup map of old subset IPs to NodeNames.
+	for n := range addresses {
+		_, ok := ipToNodeName[addresses[n].IP]
+		if !ok {
```

Comment:
what if it is found but with a different name?  I guess that was validated on create